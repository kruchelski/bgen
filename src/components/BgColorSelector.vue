<template>
  <div class="glass-container-controls main-selectors-container">
    <span class="color-selector-title">Background {{ isGradient ? 'colors' : 'color' }} </span>
    <div class="selectors-container">
      <div class="selector">
        <input
          type="color"
          class="input-color-selector"
          id="bg-color-1"
          name="bg-color-1"
          v-model="color1"
          @input="handleChange('bgColor1')"
        />
        <label class="color-name" for="bg-color-1">{{ color1 }}</label>
      </div>
      <div class="selector">
        <input
          type="color"
          class="input-color-selector"
          id="bg-color-2"
          name="bg-color-2"
          :disabled="!isGradient"
          v-model="color2"
          @input="handleChange('bgColor2')"
        />
        <label class=color-name for="bg-color-2">{{ color2 }}</label>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'BgColorSelector',

  props: {
    bgColor1: {
      type: String,
      default: '#ffffff'
    },
    bgColor2: {
      type: String,
      default: '#141414'
    }
  },

  data () {
    return {
      color1: this.bgColor1,
      color2: this.bgColor2,
      isGradient: true
    }
  },

  methods: {
    handleChange (entity) {
      const values = {
        bgColor1: this.color1,
        bgColor2: this.color2
      }
      const state = {
        entity,
        value: values[entity]
      }
      this.$emit('state-update', state)
    }
  }
}
</script>

<style scoped>
  .main-selectors-container {
    min-width: 13rem;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: stretch;
    padding: 0.25rem 1rem;
    font-size: 0.8rem;
    /* color: rgba(210, 200, 200, 0.5) */
    color: #564556
  }

  .selectors-container {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    margin-top: 2px;
  }

  .selector {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    margin: 0 0.5rem;
  }

  .glass-container-controls {
    box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.1);
    border-radius: 10px;
    background: linear-gradient(to right bottom, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.2));
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    backdrop-filter: blur(5px);
  }

  .input-color-selector {
    border: transparent;
    height: 1.5rem;
    width: 1.8rem;
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    border-radius: 2px;
    background-color: transparent;
    cursor: pointer;
  }

  .color-selector-title {
    align-self: center;
  }

  input:disabled {
    opacity: 30%;
  }

  .color-name {
    cursor: pointer;
    font-family: 'Fira Mono', monospace;
    margin-top: 0.125rem;
  }

</style>