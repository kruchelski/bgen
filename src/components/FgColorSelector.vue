<template>
  <div class="glass-container-controls main-selectors-container">
    <span class="color-selector-title">Foreground color</span>
    <div class="selectors-container">
      <div
        v-show="foreground"
        class="selector"
      >
        <input
          id="color"
          type="color"
          class="input-color-selector"
          :style="shadowColor"
          name="color"
          v-model="color"
          @input="handleChange('fgColor')"
        />
        <label class="color-name" for="color">{{ color }}</label>
      </div>
    </div>
      <button
        class="toggle-foreground-button"
        @click="toggleForeground()"
      >
        <svg v-if="!foreground" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z" />
        </svg>
        <svg v-if="foreground" xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z" />
        </svg>
        <span>{{ foreground ? 'Hide foreground' : 'Show foreground' }}</span>
      </button>
  </div>
</template>

<script>
export default {
  name: 'FgColorSelector',

  props: {
    fgColor: {
      type: String,
      default: '#ffffff'
    },

    showForeGround: {
      type: Boolean,
      default: false
    }
  },

  data () {
    return {
      color: this.fgColor,
      foreground: this.showForeGround
    }
  },

  computed: {
    shadowColor () {
      return `box-shadow: 0 0 20px ${this.color}`
    }
  },

  methods: {
    toggleForeground () {
      this.foreground = !this.foreground
      const state = {
        entity: 'showForeGround',
        value: this.foreground
      }
      this.$emit('state-update', state)
    },

    handleChange (entity) {
      const state = {
        entity,
        value: this.color
      }
      this.$emit('state-update', state)
    }
  }
}
</script>

<style scoped>
  .main-selectors-container {
    min-width: 13rem;
    min-height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: stretch;
    padding: 0.5rem 1rem;
    font-size: 0.8rem;
    color: #381838
  }

  .selectors-container {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 2px;
    margin-bottom: 0;
  }

  .selector {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    margin: 0 0.5rem;
  }

  .glass-container-controls {
    box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.1);
    border-radius: 10px;
    background: linear-gradient(to right bottom, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.2));
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    backdrop-filter: blur(5px);
  }

  .input-color-selector {
    border: transparent;
    height: 1.5rem;
    width: 1.8rem;
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    border-radius: 2px;
    background-color: transparent;
    cursor: pointer;
  }

  .color-selector-title {
    align-self: center;
  }

  .color-name {
    cursor: pointer;
    font-family: 'Fira Mono', monospace;
    margin-top: 0.125rem;
  }
  
  .toggle-foreground-button {
    position: relative;
    align-self: center;
    margin-top: 00.125rem;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: rgba(25, 19, 23, 0.5);
    color: rgba(255, 255, 255, 0.5);
    padding: 0.25rem 1rem;
    border-radius: 0.25rem;
    border: transparent;
    font-size: 0.8rem;
    font-weight: 300;
    cursor: pointer;
    transition: all 100ms;
  }

  .toggle-foreground-button:hover {
    background-color: rgba(75, 29, 75, 0.5);
    box-shadow: 0 0 20px rgba(75, 29, 75, 0.1);
  }

  .toggle-foreground-button:active {
    background-color: rgba(10, 0, 10, 0.5);
    transform: translateY(1px);
    transform: scale(0.98);
  }

  .toggle-foreground-button span {
    margin-left: 0.25rem;
  }

  .toggle-foreground-button svg {
    width: 0.8rem;
  }

  @media only screen and (max-width: 710px) {
    .main-selectors-container {
      min-width: 100%;
    }

    .selectors-container {
      display: flex;
      justify-content: space-around;
      align-items: flex-start;
      margin-top: 0.5rem;
      margin-bottom: 0.5rem;
    }
  }

</style>