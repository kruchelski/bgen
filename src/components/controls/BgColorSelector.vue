<template>
  <GlassContainer>
    <div class="main-selectors-container">
      <span class="color-selector-title">Background {{ isGradient ? 'colors' : 'color' }} </span>
      <div class="selectors-container">
        <div class="selector">
          <input
            id="bg-color-1"
            type="color"
            class="input-color-selector"
            :style="shadowColor1"
            name="bg-color-1"
            v-model="color1"
            @input="handleInputColorSelector('bgColor1')"
          />
          <input
            id="bg-color1-text"
            type="text"
            class="color-name-input"
            name="bg-color1-text"
            v-model="colorText1"
            @input="handleInputColorText('bgColor1')"
          />
        </div>
        <div v-if="isGradient" class="selector">
          <input
            id="bg-color-2"
            type="color"
            class="input-color-selector"
            :style="shadowColor2"
            name="bg-color-2"
            v-model="color2"
            @input="handleInputColorSelector('bgColor2')"
          />
          <input
            id="bg-color2-text"
            type="text"
            class="color-name-input"
            name="bg-color2-text"
            v-model="colorText2"
            @input="handleInputColorText('bgColor2')"
          />
        </div>
      </div>
      <Button @button-click="toggleGradient()">
        <svg v-if="!isGradient" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z" />
          </svg>
          <svg v-if="isGradient" xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z" />
        </svg>
        <span>{{ isGradient ? 'Disable gradient' : 'Enable gradient' }}</span>
      </Button>
    </div>
  </GlassContainer>
</template>

<script>
import Button from '@/components/common/Button'
import GlassContainer from '@/components/common/GlassContainer'
import * as colorUtils from '@/utils/colorUtils'

export default {
  name: 'BgColorSelector',

  components: {
    Button,
    GlassContainer
  },

  props: {
    bgColor1: {
      type: String,
      default: '#ffffff'
    },

    bgColor2: {
      type: String,
      default: '#141414'
    },

    gradient: {
      type: Boolean,
      default: true
    }
  },

  data () {
    return {
      color1: this.bgColor1,
      color2: this.bgColor2,
      colorText1: this.bgColor1,
      colorText2: this.bgColor2,
      isGradient: this.gradient
    }
  },

  computed: {
    shadowColor1 () {
      return `box-shadow: 0 0 20px ${this.color1}`
    },

    shadowColor2 () {
      return `box-shadow: 0 0 20px ${this.color2}`
    }
  },

  methods: {
    toggleGradient () {
      this.isGradient = !this.isGradient
      const state = {
        entity: 'gradient',
        value: this.isGradient
      }
      this.$emit('state-update', state)
    },

    handleChange (entity) {
      const values = {
        bgColor1: this.color1,
        bgColor2: this.color2
      }
      const state = {
        entity,
        value: values[entity]
      }
      this.$emit('state-update', state)
    },

    handleInputColorSelector (entity) {
      if (entity === 'bgColor1') {
        this.colorText1 = this.color1
      }
      if (entity === 'bgColor2') {
        this.colorText2 = this.color2
      }
      this.handleChange(entity)
    },

    handleInputColorText (entity) {
      let color = ''
      if (entity === 'bgColor1') {
        this.colorText1 = colorUtils.normalizeColor(this.colorText1)
        color = this.colorText1
      }
      if (entity === 'bgColor2') {
        this.colorText2 = colorUtils.normalizeColor(this.colorText2)
        color = this.colorText2
      }
      if (!colorUtils.validateInputColor(color)) return
      if (entity === 'bgColor1') {
        this.color1 = color
      } else {
        this.color2 = color
      }
      this.handleChange(entity)
    }
  }
}
</script>

<style scoped>
  .main-selectors-container {
    box-sizing: border-box;
    min-width: 15rem;
    max-width: 15rem;
    min-height: 16vh;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: stretch;
    padding: 0.5rem 1rem;
    font-size: 0.8rem;
    color: #381838;
  }

  .selectors-container {
    display: flex;
    justify-content: space-around;
    align-items: flex-start;
    margin-top: 2px;
    margin-bottom: 0;
  }

  .selector {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    margin: 0 0.5rem;
  }

  .input-color-selector {
    border: transparent;
    height: 1.5rem;
    width: 1.8rem;
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    border-radius: 2px;
    background-color: transparent;
    cursor: pointer;
  }

  .color-selector-title {
    align-self: center;
  }

  .color-name {
    cursor: pointer;
    font-family: 'Fira Mono', monospace;
    margin-top: 0.125rem;
  }

  .color-name-input {
    font-family: 'Fira Mono', monospace;
    border: 1px solid rgba(25, 19, 23, 0.5);
    height: 1.5rem;
    width: 4.5rem;
    border-radius: 0.25rem;
    background: linear-gradient(to right bottom, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.2));
    color: #381838;
    font-size: 0.8rem;
    padding: 0 0.5rem;
    box-sizing: border-box;
    margin-top: 0.25rem;
  }

  @media only screen and (max-width: 710px) {
    .main-selectors-container {
      min-width: 97vw;
    }

    .selectors-container {
      display: flex;
      justify-content: space-around;
      align-items: flex-start;
      margin-top: 0.5rem;
      margin-bottom: 0.5rem;
    }
  }

</style>