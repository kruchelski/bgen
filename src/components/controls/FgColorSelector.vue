<template>
  <div class="glass-container-controls main-selectors-container">
    <span class="color-selector-title">Foreground color</span>
    <div class="selectors-container">
      <div
        v-show="foreground"
        class="selector"
      >
        <input
          id="color"
          type="color"
          class="input-color-selector"
          :style="shadowColor"
          name="color"
          v-model="color"
          @input="handleInputColorSelector('fgColor')"
        />
        <input
          id="color-text"
          type="text"
          class="color-name-input"
          name="color-text"
          v-model="colorText"
          @input="handleInputColorText('fgColor')"
        />
      </div>
    </div>
    <Button @button-click="toggleForeground()">
      <svg v-if="!foreground" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z" />
      </svg>
      <svg v-if="foreground" xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z" />
      </svg>
      <span>{{ foreground ? 'Hide foreground' : 'Show foreground' }}</span>
    </Button>
  </div>
</template>

<script>
import Button from '@/components/common/Button'
import * as colorUtils from '@/utils/colorUtils'

export default {
  name: 'FgColorSelector',

  components: {
    Button
  },

  props: {
    fgColor: {
      type: String,
      default: '#ffffff'
    },

    showForeGround: {
      type: Boolean,
      default: false
    }
  },

  data () {
    return {
      color: this.fgColor,
      colorText: this.fgColor,
      foreground: this.showForeGround
    }
  },

  computed: {
    shadowColor () {
      return `box-shadow: 0 0 20px ${this.color}`
    }
  },

  methods: {
    toggleForeground () {
      this.foreground = !this.foreground
      const state = {
        entity: 'showForeGround',
        value: this.foreground
      }
      this.$emit('state-update', state)
    },

    handleChange (entity) {
      const state = {
        entity,
        value: this.color
      }
      this.$emit('state-update', state)
    },

    handleInputColorSelector (entity) {
      this.colorText = this.color
      this.handleChange(entity)
    },

    handleInputColorText (entity) {
      this.colorText = colorUtils.normalizeColor(this.colorText)
      const colorTemp = this.colorText
      if (!colorUtils.validateInputColor(colorTemp)) return
      this.color = colorTemp
      this.handleChange(entity)
    }
  }
}
</script>

<style scoped>
  .main-selectors-container {
    min-width: 13rem;
    min-height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: stretch;
    padding: 0.5rem 1rem;
    font-size: 0.8rem;
    color: #381838
  }

  .selectors-container {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 2px;
    margin-bottom: 0;
  }

  .selector {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    margin: 0 0.5rem;
  }

  .glass-container-controls {
    box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.1);
    border-radius: 10px;
    background: linear-gradient(to right bottom, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.2));
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    backdrop-filter: blur(5px);
  }

  .input-color-selector {
    border: transparent;
    height: 1.5rem;
    width: 1.8rem;
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    border-radius: 2px;
    background-color: transparent;
    cursor: pointer;
  }

  .color-selector-title {
    align-self: center;
  }

  .color-name-input {
    font-family: 'Fira Mono', monospace;
    border: 1px solid rgba(25, 19, 23, 0.5);
    height: 1.5rem;
    width: 4.5rem;
    border-radius: 0.25rem;
    background: linear-gradient(to right bottom, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.2));
    color: #381838;
    font-size: 0.8rem;
    padding: 0 0.5rem;
    box-sizing: border-box;
    margin-top: 0.25rem;
  }
  
  @media only screen and (max-width: 710px) {
    .main-selectors-container {
      min-width: 100%;
    }

    .selectors-container {
      display: flex;
      justify-content: space-around;
      align-items: flex-start;
      margin-top: 0.5rem;
      margin-bottom: 0.5rem;
    }
  }

</style>