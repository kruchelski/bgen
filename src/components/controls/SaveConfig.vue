<template>
  <div class="glass-container-controls main-input-container">
    <span class="save-config-title">Save config</span>
    <div class="input-container">
      <div
        class="input-box"
      >        
        <svg v-if="errorState" class="icon-error" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z" />
        </svg>
        <input
          id="configName"
          type="text"
          class="input-text"
          :class="errorState ? 'input-text-error' : ''"
          name="configName"
          placeholder="Config name"
          v-model="configName"
          @keyup.enter="saveConfig(configName, configId)"
        />
      </div>
    </div>
      <button
        class="save-button"
        @click="saveConfig(configName, configId)"
      >
        <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M8 7H5a2 2 0 00-2 2v9a2 2 0 002 2h14a2 2 0 002-2V9a2 2 0 00-2-2h-3m-1 4l-3 3m0 0l-3-3m3 3V4" />
        </svg>
        <span>{{ configExists ? 'Update config' : 'Save config' }}</span>
      </button>
  </div>
</template>

<script>
import * as ToastrService from '@/services/ToastrService'

export default {
  name: 'SaveConfig',

  props: {
    id: {
      type: Number,
      default: null
    },

    name: {
      type: String,
      default: ''
    },

    configs: {
      type: Array,
      default: () => []
    }
  },

  data () {
    return {
      configId: this.id,
      configName: this.name,
      hasError: false
    }
  },

  computed: {
    errorState () {
      if (this.hasError && !this.configName) {
        return true
      }
      return false
    },
  
    configExists () {
      if (this.configId) return true
      return false
    }
  },

  watch: {
    configName () {
      const config = this.configs.find(config => this.configName === config.name)
      if (config) {
        this.configId = config.id
      } else {
        this.configId = null
      }
    }
  },

  methods: {
    saveConfig (name, id) {
      if (this.configNameHasError(name)) {
        this.hasError = true
        ToastrService.displayToastr('Error saving: Config name cannot be empty', 'Error saving', 'fail')
        return
      }
      this.hasError = false
      const saveEvent = {
        name,
        id
      }
      this.$emit('config-save', saveEvent)
      this.configId = null
      this.configName = ''
      this.hasError = false
    },

    configNameHasError (name) {
      if (!name) return true
      return false
    }
  }
}
</script>

<style scoped>
  .main-input-container {
    min-width: 13rem;
    min-height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: stretch;
    padding: 0.5rem 0.5rem;
    font-size: 0.8rem;
    color: #381838
  }

  .input-container {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 2px;
    margin-bottom: 0;
  }

  .input-box {
    display: flex;
    /* flex-direction: column; */
    justify-content: flex-start;
    align-items: center;
    margin: 0 0.5rem;
    width: 10.5rem;
    box-sizing: border-box;
  }

  .glass-container-controls {
    box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.1);
    border-radius: 10px;
    background: linear-gradient(to right bottom, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.2));
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    backdrop-filter: blur(5px);
  }

  .input-text {
    border: 1px solid rgba(25, 19, 23, 0.5);
    height: 1.5rem;
    width: 100%;

    border-radius: 0.25rem;
    background: linear-gradient(to right bottom, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.2));
    color: #381838;
    font-size: 0.8rem;
    padding: 0 0.5rem;
    box-sizing: border-box;
  }

  .input-text:focus {
    outline: 2px solid #783878;
  }

  .input-text-error {
    border: 1px solid #8b0000;
  }

  .input-text.input-text-error:focus {
    outline: 2px solid #8b0000;
  }

  .save-config-title {
    align-self: center;
  }

  .icon-error {
    width: 1.3rem;
    color: #8b0000;
    margin-right: 0.25rem;
  }
  
  .save-button {
    position: relative;
    align-self: center;
    margin-top: 00.125rem;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: rgba(25, 19, 23, 0.5);
    color: rgba(255, 255, 255, 0.5);
    padding: 0.25rem 1rem;
    border-radius: 0.25rem;
    border: transparent;
    font-size: 0.8rem;
    font-weight: 300;
    cursor: pointer;
    transition: all 100ms;
  }

  .save-button:hover {
    background-color: rgba(75, 29, 75, 0.5);
    box-shadow: 0 0 20px rgba(75, 29, 75, 0.1);
  }

  .save-button:active {
    background-color: rgba(10, 0, 10, 0.5);
    transform: translateY(1px);
    transform: scale(0.98);
  }

  .save-button span {
    margin-left: 0.25rem;
  }

  .save-button svg {
    width: 0.8rem;
  }

  @media only screen and (max-width: 710px) {
    .main-input-container {
      min-width: 100%;
    }

    .input-container {
      display: flex;
      justify-content: space-around;
      align-items: flex-start;
      margin-top: 0.5rem;
      margin-bottom: 0.5rem;
    }
  }

</style>