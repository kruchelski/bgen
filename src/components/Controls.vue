<template>
  <div class="controls-container">
    <div class="main-controls">
      <div class="glass-container control">
        AiAi
      </div>
      <div class="control">
        <BgColorSelector
          :bgColor1="bgColor1"
          :bgColor2="bgColor2"
          @state-update="handleStateUpdate"
        />
      </div>
      <div class="glass-container control">
        Aiai
      </div>
    </div>
  </div>
</template>

<script>
import BgColorSelector from '@/components/BgColorSelector'

export default {
  name: 'Controls',

  components: {
    BgColorSelector
  },

  props: {
    bgColor1: {
      type: String,
      default: '#ffffff'
    },

    bgColor2: {
      type: String,
      default: '#141414'
    },

    showForeGround: {
      type: Boolean,
      default: false
    },

    fgColor: {
      type: String,
      defualt: '#ffffff'
    }
  },

  methods: {
    handleStateUpdate (state) {
      this.$emit('state-update', state)
    }
  }
}
</script>

<style scoped>
  .glass-container {
    box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.1);
    border-radius: 1rem;
    background: linear-gradient(to right bottom, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.2));
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    border-left: 1px solid rgba(255, 255, 255, 0.5);
    backdrop-filter: blur(5px);
  }

  .controls-container {
    min-width: 97vw;
    max-width: 97vw;
    min-height: 13vh;
    margin: 0 auto;
    padding: 0.5rem 1rem;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: stretch;
    box-sizing: border-box;
  }

  .main-controls {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
  }

  .control {
    flex: 1;
    display: flex;
    justify-content: center;
    align-items: flex-start;
  }

  @media only screen and (max-width: 710px) {
    .main-controls {
      display: flex;
      flex-direction: column;
      justify-content: flex-start;
      align-items: stretch;
    }

    .main-controls :nth-child(2) {
      order: -1;
    }

    .control {
      flex: 0;
      display: flex;
      justify-content: center;
      align-items: flex-start;
      margin: 0.5rem 0;
    }
  }
</style>