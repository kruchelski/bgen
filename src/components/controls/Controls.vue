<template>
  <div class="controls-container">
    <div class="control">
      <FgColorSelector
        :fgColor="fgColor"
        :showForeGround="showForeGround"
        @state-update="handleStateUpdate"
      />
    </div>
    <div class="control">
      <BgColorSelector
        :bgColor1="bgColor1"
        :bgColor2="bgColor2"
        :gradient="gradient"
        @state-update="handleStateUpdate"
      />
    </div>
    <div class="control">
      <SaveConfig
        :id="id"
        :name="name"
        :configs="configs"
        @config-save="handleConfigSave"
      />
    </div>
  </div>
</template>

<script>
import BgColorSelector from '@/components/controls/BgColorSelector'
import FgColorSelector from '@/components/controls/FgColorSelector'
import SaveConfig from '@/components/controls/SaveConfig'

export default {
  name: 'Controls',

  components: {
    BgColorSelector,
    FgColorSelector,
    SaveConfig
  },

  props: {
    bgColor1: {
      type: String,
      default: '#ffffff'
    },

    bgColor2: {
      type: String,
      default: '#141414'
    },

    gradient: {
      type: Boolean,
      default: true
    },

    showForeGround: {
      type: Boolean,
      default: false
    },

    fgColor: {
      type: String,
      default: '#ffffff'
    },

    id: {
      type: Number,
      default: null,
    },

    name: {
      type: String,
      default: ''
    },

    configs: {
      type: Array,
      default: () => []
    }
  },

  methods: {
    handleStateUpdate (state) {
      this.$emit('state-update', state)
    },

    handleConfigSave (config) {
      this.$emit('config-save', config)
    }
  }
}
</script>

<style scoped>
  .controls-container {
    min-width: 97vw;
    max-width: 97vw;
    min-height: 16vh;
    margin: 0 auto;
    padding: 0.5rem 1rem;
    display: flex;
    justify-content: space-between;
    align-items: stretch;
    box-sizing: border-box;
  }

  .control {
    flex: 1;
    display: flex;
    justify-content: center;
    align-items: flex-start;
  }

  @media only screen and (max-width: 710px) {
    .controls-container {
      display: flex;
      flex-direction: column;
      justify-content: flex-start;
      align-items: stretch;
    }

    .control:nth-child(2) {
      order: 0;
    }

    .control:nth-child(1) {
      order: 1;
    }

    .control:nth-child(3) {
      order: 1;
    }

    .control {
      flex: 0;
      display: flex;
      justify-content: center;
      align-items: flex-start;
      margin: 0.5rem 0;
    }
  }
</style>