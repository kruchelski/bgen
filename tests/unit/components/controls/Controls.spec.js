import Controls from "@/components/controls/Controls";
import BgColorSelector from '@/components/controls/BgColorSelector'
import FgColorSelector from '@/components/controls/FgColorSelector'
import SaveConfig from '@/components/controls/SaveConfig'
import { createLocalVue, shallowMount } from "@vue/test-utils";
import VueTilt from "vue-tilt.js";

const localVue = createLocalVue();
localVue.use(VueTilt);

const build = (testMount = {}) => {
  const { testData = null, testMethods = null } = testMount;
  let mountOptions = {};

  if (testData) {
    mountOptions = {
      sync: false,
      localVue,
      data() {
        return testData;
      }
    };
  } else {
    mountOptions = {
      sync: false,
      localVue
    };
  }

  if (testMethods) {
    mountOptions.methods = testMethods;
  }

  return shallowMount(Controls, mountOptions);
};

describe('Tests Controls render', () => {
  beforeEach(() => jest.clearAllMocks());

  it('should render it', () => {
    const wrapper = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render BgColorSelector component', () => {
    const wrapper = build()
    const childPlaceholderContent = wrapper.findComponent(BgColorSelector)
    expect(childPlaceholderContent.exists()).toBe(true)
  })

  it('should render FgColorSelector component', () => {
    const wrapper = build()
    const childControls = wrapper.findComponent(FgColorSelector)
    expect(childControls.exists()).toBe(true)
  })

  it('should render SaveConfig component', () => {
    const wrapper = build()
    const childControls = wrapper.findComponent(SaveConfig)
    expect(childControls.exists()).toBe(true)
  })
});

describe('Tests handleStateUpdate', () => {
  beforeEach(() => jest.clearAllMocks())

  it('should emit an state-update event with the provided data', () => {
    const mockState = {
      entity: 'testEntity',
      value: 'testValue'
    }
    const wrapper = build()
    wrapper.vm.handleStateUpdate(mockState)
    expect(wrapper.emitted(['state-update'])[0][0]).toEqual(mockState)
  })
})

describe('Tests handleConfigSave', () => {
  beforeEach(() => jest.clearAllMocks())

  it('should emit an config-save event with the provided data', () => {
    const mockConfig = {
      name: 'testName',
      id: 123
    }
    const wrapper = build()
    wrapper.vm.handleConfigSave(mockConfig)
    expect(wrapper.emitted(['config-save'])[0][0]).toEqual(mockConfig)
  })
})