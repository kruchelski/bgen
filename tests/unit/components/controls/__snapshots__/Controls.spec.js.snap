// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Controls render should render it 1`] = `
<div class="controls-container">
  <div class="control">
    <fgcolorselector-stub fgcolor="#ffffff"></fgcolorselector-stub>
  </div>
  <div class="control">
    <bgcolorselector-stub bgcolor1="#ffffff" bgcolor2="#141414" gradient="true"></bgcolorselector-stub>
  </div>
  <div class="control">
    <saveconfig-stub name="" configs=""></saveconfig-stub>
  </div>
</div>
`;
